src/features/calendar-events/index.js

```diff
@@ -1,33 +1,103 @@
-import React from 'react'
-import { Calendar } from 'react-big-calendar'
+import React, { useState, useEffect } from 'react'
+import { Spin } from 'antd'
 
+import EventModal from './calendar-modal'
 import config from './calendarConfig'
+import {
+  CalendarEventsContainer,
+  CurrentEventsNumber,
+  StyledCalendar,
+} from './styled-components'
 
-import 'react-big-calendar/lib/css/react-big-calendar.css'
+const {
+  spinSize,
+  defaultDate,
+  drilldownView,
+  defaultView,
+  localizer,
+  selectable,
+  resizable,
+  tempEvents,
+} = config
 
 const CalendarRBC = () => {
-  const {
-    defaultDate,
-    drilldownView,
-    defaultView,
-    events,
-    localizer,
-    resizable,
-    style,
-  } = config
+  const [isLoading, setIsLoading] = useState(true)
+  const [modalVisibility, setModalVisibility] = useState(false)
+  const [inputValue, setInputValue] = useState('')
+  const [currentEvent, setCurrentEvent] = useState({})
+  const [calendarEvents, setCalendarEvents] = useState(
+    JSON.parse(localStorage.getItem('CalendarStorageEvents')) || tempEvents,
+  )
+
+  useEffect(() => {
+    localStorage.setItem(
+      'CalendarStorageEvents',
+      JSON.stringify(calendarEvents),
+    )
+  }, [calendarEvents])
+
+  setTimeout(() => {
+    setIsLoading(false)
+  }, 750)
+
+  const handleInput = e => {
+    setInputValue(e.target.value)
+  }
+
+  const handleModalOpen = ({ start, end }) => {
+    if (start.getDate() >= defaultDate.getDate()) {
+      setModalVisibility(true)
+      setCurrentEvent({ start, end })
+    }
+  }
+
+  const handleOk = () => {
+    setModalVisibility(false)
+    if (inputValue) {
+      setCurrentEvent(Object.assign(currentEvent, { title: inputValue }))
```

Comment:
You should not mutate state variables directly, use either `setCurrentEvent(Object.assign({}, currentEvent, { title: inputValue }))` or to be more consistent with the rest of the code, use the **spread** operator `setCurrentEvent({...currentEvent, title: inputValue})`.